typing/ctype.ml

```diff
@@ -4533,24 +4533,39 @@ let same_constr env t1 t2 =
 let () =
   Env.same_constr := same_constr
 
-let maybe_pointer_type env typ =
+let rec get_type_repr env typ =
    match (repr typ).desc with
   | Tconstr(p, _args, _abbrev) ->
     begin try
       let type_decl = Env.find_type p env in
-      not type_decl.type_immediate
-    with Not_found -> true
+      type_decl.type_repr
+    with Not_found -> Repr_any
     (* This can happen due to e.g. missing -I options,
        causing some .cmi files to be unavailable.
        Maybe we should emit a warning. *)
     end
   | Tvariant row ->
       let row = Btype.row_repr row in
-      (* if all labels are devoid of arguments, not a pointer *)
```

Comment:
The code might be fairly straightforward, but I'm still sad that we're losing this comment.

In particular, I feel like this comment made it vaguely easier to understand why we're checking `row_closed`.